milli/src/update/settings.rs

```diff
@@ -210,6 +220,35 @@ impl<'a, 't, 'u, 'i> Settings<'a, 't, 'u, 'i> {
         Ok(true)
     }
 
+    fn update_stop_words(&mut self) -> anyhow::Result<bool> {
+        match self.stop_words {
+            Some(Some(ref stop_words)) => {
+                let current = self.index.stop_words(self.wtxn)?;
+                let identical = current
+                    .map(|current| {
+                        stop_words.iter().all(|elem| current.contains(elem))
+                            && stop_words.len() == current.len()
+                    })
+                    .unwrap_or(false);
+
+                // we want to re-create our fst
+                if !identical {
+                    // we need to sort it first. TODO: use btreemap
+                    let mut words: Vec<String> = stop_words.iter().cloned().collect();
+                    words.sort();
+                    let fst = fst::Set::from_iter(&words)?;
+                    self.index.put_stop_words(self.wtxn, &fst)?;
+                }
+                Ok(!identical)
```

Comment:
I am not that's most visible way of doing this, maybe constructing the FST first and then comparing the internal bytes for equality (`fst.as_fst().as_bytes()`).

And to simplify the FST creation why can't we just ask for a `BTreeSet` instead of an `HashSet`?